src/iminuit/cost.py

```diff
@@ -1153,6 +1154,11 @@ def _pred(self, args: Sequence[float]) -> NDArray:
         # differences can come out negative due to round-off error in subtraction,
         # we set negative values to zero
         d[d < 0] = 0
+        if len(self._data) != len(d):
+            raise ValueError(
+                f"Expected model to return an array of size {len(self._data)},"
+                f"but it returns an array of size {len(d)}"
+            )
```

Comment:
Good, but `d` and `self._data` could be multidimensional, so we want to check `self._data.shape == d.shape`.

Reply:
The error message needs to be adapted according, "array of size" -> "array with shape" etc